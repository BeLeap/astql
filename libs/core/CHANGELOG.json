{
  "name": "@astql/core",
  "entries": [
    {
      "version": "0.1.4",
      "tag": "@astql/core_v0.1.4",
      "date": "Mon, 01 Nov 2021 13:02:58 GMT",
      "comments": {
        "patch": [
          {
            "comment": "add info"
          }
        ],
        "dependency": [
          {
            "comment": "Updating dependency \"@astql/languages\" from `0.1.3` to `0.1.4`"
          },
          {
            "comment": "Updating dependency \"@astql/query\" from `0.1.3` to `0.1.4`"
          },
          {
            "comment": "Updating dependency \"@astql/parser\" from `0.1.3` to `0.1.4`"
          },
          {
            "comment": "Updating dependency \"@astql/traverse\" from `0.1.3` to `0.1.4`"
          },
          {
            "comment": "Updating dependency \"@astql/js.typescript\" from `~0.1.1` to `~0.1.2`"
          }
        ]
      }
    },
    {
      "version": "0.1.3",
      "tag": "@astql/core_v0.1.3",
      "date": "Sat, 30 Oct 2021 06:49:29 GMT",
      "comments": {
        "patch": [
          {
            "comment": "add features"
          }
        ],
        "dependency": [
          {
            "comment": "Updating dependency \"@astql/languages\" from `0.1.2` to `0.1.3`"
          },
          {
            "comment": "Updating dependency \"@astql/query\" from `0.1.2` to `0.1.3`"
          },
          {
            "comment": "Updating dependency \"@astql/parser\" from `0.1.2` to `0.1.3`"
          },
          {
            "comment": "Updating dependency \"@astql/traverse\" from `0.1.2` to `0.1.3`"
          }
        ]
      }
    }
  ]
}
