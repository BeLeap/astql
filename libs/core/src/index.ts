export { default as Code } from './Code';
export type { Selector } from './Code';
export { default as Project } from './Project';
export type { default as ParserConfig } from './ParserConfig';
export { default as defaultParserInterface } from './utils/defaultParserInterface';
export { default as generateVisitorKeys } from './utils/generateVisitorKeys';
export { default as multipleRequire } from './utils/multiple-require';
export * from '@astql/query';
import * as languages from '@astql/languages';
export { languages };
