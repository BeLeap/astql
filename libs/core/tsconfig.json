{
  "extends": "@digigov/cli-build/tsconfig.base",
  "include": [
    "./src/**/*.ts",
    "./src/**/*.js"
  ],
  "exclude": [
    "node_modules"
  ],
  "compilerOptions": {
    "types": ["jest", "node"],
    "baseUrl": ".",
    "paths": {
      "@astql/languages": ["../languages/src"],
      "@astql/query": ["../query/src"],
      "@astql/parser": ["../parser/src"]
    },
    "skipLibCheck": true,
    "strict": false,
    "esModuleInterop": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "outDir": "dist"
  }
}