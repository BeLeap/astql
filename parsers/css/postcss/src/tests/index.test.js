import { Code } from '@astql/core';

import codeExample from './codeExample.js';

const code = new Code('file.css', codeExample, {
  parser: require('../index.js').default,
});
it('code should parse without error', async () => {
  await code.parse();
  expect(code.ast).toHaveProperty('_type');
});
it('ast.getFullText() should match codeExample.txt', async () => {
  expect(code.ast.getFullText()).toMatchSnapshot();
});
it('should query generated ast', async () => {
  const result = await code.query('*');
  expect(result.length).not.toBe(0);
  expect(result.length).toMatchSnapshot();
});
