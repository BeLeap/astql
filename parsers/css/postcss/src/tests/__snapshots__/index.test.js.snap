// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should query generated ast 1`] = `9`;
